keps/sig-network/3766-referencegrant/README.md

```diff
@@ -0,0 +1,492 @@
+[x] Multiple conformant implementations.
+[x] Widespread implementation and usage.
+[ ] Conformance tests that exercise all ReferenceGrant API calls (not the actual implementation of the API).
+[ ] At least 6 months of soak time as a beta API (graduated to beta in Dec 2022).
+
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+Version skew is a bit different here since it won't be implemented in-tree.
+There will be some implementation that support both the API defined by Gateway
+API and this API. Since these resources are entirely additive and can be
+duplicative, we can copy Gateway API resources to the new API group and delete
+the old Gateway API resources as part of a seamless migration. We expect that
+many implementations will provide this recommendation to users, and we may even
+provide tooling to simplify this process.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Other
+  - Describe the mechanism: Enable beta ReferenceGrant API
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+    No
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+    No
+
+###### Does enabling the feature change any default behavior?
+
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The API would become accessible again, implementing controllers may need to be
+restarted to pick up the presence of this API.
+
+###### Are there any tests for feature enablement/disablement?
+
+No
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+API enablement may not work, but that would not be unique to this API.
+
+###### What specific metrics should inform a rollback?
+
+N/A, this is just an API
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+N/A, this is just an API
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No
+
+### Monitoring Requirements
+
+N/A, this is just an API
+
+###### How can an operator determine if the feature is in use by workloads?
+```
+kubectl get referencegrants --all-namespaces
+```
+
+###### How can someone using this feature know that it is working for their instance?
+
+This will be dependent on the API that ReferenceGrant is used with. In Gateway API,
+each resource has clear status conditions that reflect the validity of a cross-namespace
+reference.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+N/A, this is just an API
```

Comment:
if there's a library provided, and expectation of reacting to revocation of reference grants, a possible SLO would be time between revocation and notification by the library

Reply:
Good call, added.